backend-public/contracts/bns2.clar

```diff
@@ -1,123 +1,64 @@
+(define-constant err-not-authorized (err u403))
 
-;; bns2
-;;
-(define-data-var owned-namespace (buff 20) 0x67676767676767676767)
-(define-data-var owner principal tx-sender)
+(define-data-var owned-namespace (buff 20) 0x)
+(define-data-var contract-owner principal tx-sender)
 
 (define-public (namespace-preorder (hashed-salted-namespace (buff 20))
                                    (stx-to-burn uint))
-    (let ((this-ctr (as-contract tx-sender)))
-        (asserts! (is-authorized tx-sender) err-not-authorized)
-        (try! (stx-transfer? stx-to-burn tx-sender this-ctr))
+    (begin
+        (try! (is-contract-owner))
+        (try! (stx-transfer? stx-to-burn tx-sender (as-contract tx-sender)))
         (to-uint-response (as-contract (contract-call? 'SP000000000000000000002Q6VF78.bns
         namespace-preorder hashed-salted-namespace stx-to-burn)))))
```

Comment:
Have you thought about moving this call to `name-register` function? That way you'll have to submit only one TX to register the name.

Reply:
It comes at the risk that other users front run the registration if the name is broadcast in plain text

Reply:
Fair point.

Reply:
At the same time, if the owner of this contract controls registration then we could have a single function to immediately register. I think it's worth having both.